Vice_xvic_libretro problems

I've compiled the new version of xvic vice on linux arm board but it has some problems, it doesnt load zipped roms and when i try to overrides core option retroarch give me a generric error saving core option. If i compile x64 i don't have this problem.

This is the verbose log I obtain triyng to load a zipped file 👍 

[INFO] === Build =======================================
[INFO] Capabilities:  NEON VFPv3
[INFO] Built: May  5 2021
[INFO] Version: 1.9.2
[INFO] =================================================
[INFO] [Input]: Found input driver: "sdl".
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] Version of libretro API: 1
[INFO] Compiled against API: 1
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Video @ 384x224
[INFO] [EGL] Falling back to eglGetDisplay
[INFO] [EGL]: EGL version: 1.4
[INFO] [GL]: Found GL context: fbdev_mali
[INFO] [GL]: Detecting screen resolution 0x0.
[INFO] [EGL]: Current context: 0x40000001.
[INFO] [GL]: Vendor: ARM, Renderer: Mali-400 MP.
[INFO] [GL]: Version: OpenGL ES 2.0.
[INFO] [GL]: Using resolution 384x224
[INFO] [GL]: Default shader backend found: glsl.
[INFO] [Shader driver]: Using GLSL shader backend.
[WARN] [GL]: Stock GLSL shaders will be used.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GL]: Using 4 textures.
[INFO] [GL]: Loaded 1 program(s).
[INFO] [Font]: Using font rendering backend: bitmap.
[INFO] [Video]: Graphics driver did not initialize an input driver. Attempting to pick a suitable driver.
[INFO] [Joypad]: Found joypad driver: "sdl".
[INFO] [Video]: Found display server: null
[INFO] ALSA: Using floating point format.
[INFO] ALSA: Period size: 512 frames
[INFO] ALSA: Buffer size: 1536 frames
[INFO] [Display]: Found display driver: "gl".
[ERROR] [Font]: Failed to create rendering backend: stb-unicode.
[INFO] [Font]: Using font rendering backend: bitmap.
[ERROR] [Font]: Failed to create rendering backend: stb-unicode.
[INFO] [Font]: Using font rendering backend: bitmap.
[ERROR] [Font]: Failed to create rendering backend: stb-unicode.
[INFO] [Font]: Using font rendering backend: bitmap.
[ERROR] [MIDI]: snd_seq_open failed with error -2.
[ERROR] [MIDI]: Initialization failed.
[INFO] [SRAM]: SRAM will not be saved.
[INFO] [Playlist]: Loading history file: [/.config/retroarch/content_history.lpl].
[INFO] [Playlist]: Loading history file: [/.config/retroarch/content_music_history.lpl].
[INFO] [Playlist]: Loading history file: [/.config/retroarch/content_image_history.lpl].
[INFO] [Playlist]: Loading favorites file: [/.config/retroarch/content_favorites.lpl].
[INFO] [GL]: VSync => ON
[INFO] [EGL]: eglSwapInterval(1)
[INFO] [GL]: VSync => ON
[INFO] [EGL]: eglSwapInterval(1)
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [CONTENT LOAD]: Updating firmware status for: /storage/cores/vice_xvic35_libretro.so on /storage/system
[INFO] [CORE]: Using content: /storage/roms/Commodore - vic20/Commodore VIC20 - Games - [PRG]/Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].zip.
[INFO] [Core]: Content ran for a total of: 00 hours, 00 minutes, 00 seconds.
[INFO] [Core]: Unloading core..
[INFO] [Core]: Unloading core symbols..
[INFO] [Video]: Does not have enough samples for monitor refresh rate estimation. Requires to run for at least 4096 frames.
[INFO] === Build =======================================
[INFO] Capabilities:  NEON VFPv3
[INFO] Built: May  5 2021
[INFO] Version: 1.9.2
[INFO] =================================================
[INFO] [Input]: Found input driver: "sdl".
[INFO] [Core]: Loading dynamic libretro core from: "/storage/cores/vice_xvic35_libretro.so"
[INFO] [Overrides]: Core-specific overrides found at "/.config/retroarch/config/VICE xvic/VICE xvic.cfg".
[INFO] [Overrides]: No content-dir-specific overrides found at "/.config/retroarch/config/VICE xvic/Commodore VIC20 - Games - [PRG].cfg".
[INFO] [Overrides]: No game-specific overrides found at "/.config/retroarch/config/VICE xvic/Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].cfg".
[INFO] [Config]: Appending config "/.config/retroarch/config/VICE xvic/VICE xvic.cfg".
[INFO] [Environ]: SYSTEM_DIRECTORY: "/storage/system".
[INFO] [Environ]: CORE_ASSETS_DIRECTORY: "/tmp".
[INFO] [Environ]: GET_SAVE_DIRECTORY.
[INFO] [Environ]: SET_CONTROLLER_INFO.
[INFO] Controller port: 1
[INFO]    Joystick (ID: 513)
[INFO]    Keyboard (ID: 259)
[INFO]    None (ID: 0)
[INFO] Controller port: 2
[INFO]    Joystick (ID: 513)
[INFO]    Keyboard (ID: 259)
[INFO]    None (ID: 0)
[INFO] Controller port: 3
[INFO]    Joystick (ID: 513)
[INFO]    Keyboard (ID: 259)
[INFO]    None (ID: 0)
[INFO] Controller port: 4
[INFO]    Joystick (ID: 513)
[INFO]    Keyboard (ID: 259)
[INFO]    None (ID: 0)
[INFO] Controller port: 5
[INFO]    Keyboard (ID: 259)
[INFO]    None (ID: 0)
[INFO] [Environ]: GET_CORE_OPTIONS_VERSION.
[INFO] [Environ]: SET_CORE_OPTIONS.
[INFO] [Environ]: SET_SUPPORT_NO_GAME: yes.
[INFO] [Remaps]: Remap directory: "/.config/retroarch/config/remaps".
[INFO] [Remaps]: Core-specific remap found at "/.config/retroarch/config/remaps/VICE xvic/VICE xvic.rmp".
[INFO] [Overrides]: Redirecting save file to "/storage/savefiles/Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].srm".
[INFO] [Overrides]: Redirecting save state to "/storage/savestates/Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].state".
[INFO] [Environ]: GET_LOG_INTERFACE.
[INFO] [Environ]: GET_PERF_INTERFACE.
[INFO] [Environ]: SYSTEM_DIRECTORY: "/storage/system".
[INFO] [Environ]: CORE_ASSETS_DIRECTORY: "/tmp".
[INFO] [Environ]: GET_DISK_CONTROL_INTERFACE_VERSION.
[INFO] [Environ]: SET_DISK_CONTROL_EXT_INTERFACE.
[INFO] [Environ]: SET_SERIALIZATION_QUIRKS.
[INFO] [Environ]: SET_INPUT_DESCRIPTORS:
[INFO] 	RetroPad, Port 1, Button "B (bottom)" => "B / Fire"
[INFO] 	RetroPad, Port 1, Button "Y (left)" => "Y"
[INFO] 	RetroPad, Port 1, Button "Select" => "Select"
[INFO] 	RetroPad, Port 1, Button "Start" => "Start"
[INFO] 	RetroPad, Port 1, Button "D-Pad Up" => "Up"
[INFO] 	RetroPad, Port 1, Button "D-Pad Down" => "Down"
[INFO] 	RetroPad, Port 1, Button "D-Pad Left" => "Left"
[INFO] 	RetroPad, Port 1, Button "D-Pad Right" => "Right"
[INFO] 	RetroPad, Port 1, Button "A (right)" => "A"
[INFO] 	RetroPad, Port 1, Button "X (up)" => "X"
[INFO] 	RetroPad, Port 1, Button "L" => "L"
[INFO] 	RetroPad, Port 1, Button "R" => "R"
[INFO] 	RetroPad, Port 1, Button "L2" => "L2"
[INFO] 	RetroPad, Port 1, Button "R2" => "R2"
[INFO] 	RetroPad, Port 1, Button "L3" => "L3"
[INFO] 	RetroPad, Port 1, Button "R3" => "R3"
[INFO] 	RetroPad, Port 2, Button "B (bottom)" => "B / Fire"
[INFO] 	RetroPad, Port 2, Button "Y (left)" => "Y"
[INFO] 	RetroPad, Port 2, Button "Select" => "Select"
[INFO] 	RetroPad, Port 2, Button "Start" => "Start"
[INFO] 	RetroPad, Port 2, Button "D-Pad Up" => "Up"
[INFO] 	RetroPad, Port 2, Button "D-Pad Down" => "Down"
[INFO] 	RetroPad, Port 2, Button "D-Pad Left" => "Left"
[INFO] 	RetroPad, Port 2, Button "D-Pad Right" => "Right"
[INFO] 	RetroPad, Port 2, Button "A (right)" => "A"
[INFO] 	RetroPad, Port 2, Button "X (up)" => "X"
[INFO] 	RetroPad, Port 2, Button "L" => "L"
[INFO] 	RetroPad, Port 2, Button "R" => "R"
[INFO] 	RetroPad, Port 2, Button "L2" => "L2"
[INFO] 	RetroPad, Port 2, Button "R2" => "R2"
[INFO] 	RetroPad, Port 2, Button "L3" => "L3"
[INFO] 	RetroPad, Port 2, Button "R3" => "R3"
[INFO] [Environ]: SET_KEYBOARD_CALLBACK.
[INFO] [Environ]: SET_PIXEL_FORMAT: RGB565.
[INFO] [Environ]: GET_VARIABLE vice_cartridge:
	none
[INFO] [Environ]: GET_VARIABLE vice_autostart:
	enabled
[INFO] [Environ]: GET_VARIABLE vice_autoloadwarp:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_floppy_write_protection:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_work_disk:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_virtual_device_traps:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_drive_true_emulation:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_drive_sound_emulation:
	20%
[INFO] [Environ]: GET_VARIABLE vice_datasette_sound:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_audio_leak_emulation:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_sound_sample_rate:
	48000
[INFO] [Environ]: GET_VARIABLE vice_vic20_model:
	VIC20 PAL auto
[INFO] [Environ]: GET_VARIABLE vice_vic20_memory_expansions:
	none
[INFO] [Environ]: GET_VARIABLE vice_zoom_mode:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_zoom_mode_crop:
	both
[INFO] [Environ]: GET_VARIABLE vice_aspect_ratio:
	auto
[INFO] [Environ]: GET_VARIABLE vice_manual_crop_top:
	0
[INFO] [Environ]: GET_VARIABLE vice_manual_crop_bottom:
	0
[INFO] [Environ]: GET_VARIABLE vice_manual_crop_left:
	0
[INFO] [Environ]: GET_VARIABLE vice_manual_crop_right:
	0
[INFO] [Environ]: GET_VARIABLE vice_gfx_colors:
	16bit
[INFO] [Environ]: GET_VARIABLE vice_vic_filter:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_vic20_external_palette:
	colodore_vic
[INFO] [Environ]: GET_VARIABLE vice_vic_color_gamma:
	2800
[INFO] [Environ]: GET_VARIABLE vice_vic_color_tint:
	1000
[INFO] [Environ]: GET_VARIABLE vice_vic_color_saturation:
	1000
[INFO] [Environ]: GET_VARIABLE vice_vic_color_contrast:
	1000
[INFO] [Environ]: GET_VARIABLE vice_vic_color_brightness:
	1000
[INFO] [Environ]: GET_VARIABLE vice_userport_joytype:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_joyport_type:
	1
[INFO] [Environ]: GET_VARIABLE vice_joyport_pointer_color:
	blue
[INFO] [Environ]: GET_VARIABLE vice_analogmouse_deadzone:
	20
[INFO] [Environ]: GET_VARIABLE vice_analogmouse_speed:
	1.0
[INFO] [Environ]: GET_VARIABLE vice_dpadmouse_speed:
	6
[INFO] [Environ]: GET_VARIABLE vice_mouse_speed:
	100
[INFO] [Environ]: GET_VARIABLE vice_keyrah_keypad_mappings:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_keyboard_keymap:
	positional
[INFO] [Environ]: GET_VARIABLE vice_physical_keyboard_pass_through:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_retropad_options:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_turbo_fire:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_turbo_fire_button:
	B
[INFO] [Environ]: GET_VARIABLE vice_turbo_pulse:
	6
[INFO] [Environ]: GET_VARIABLE vice_reset:
	autostart
[INFO] [Environ]: GET_VARIABLE vice_vkbd_theme:
	auto
[INFO] [Environ]: GET_VARIABLE vice_vkbd_transparency:
	25%
[INFO] [Environ]: GET_VARIABLE vice_statusbar:
	bottom
[INFO] [Environ]: GET_VARIABLE vice_mapping_options_display:
	enabled
[INFO] [Environ]: GET_VARIABLE vice_audio_options_display:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_video_options_display:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_read_vicerc:
	enabled
[INFO] [Environ]: GET_VARIABLE vice_mapper_select:
	TOGGLE_VKBD
[INFO] [Environ]: GET_VARIABLE vice_mapper_start:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_b:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_a:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_y:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_x:
	RETROK_SPACE
[INFO] [Environ]: GET_VARIABLE vice_mapper_l:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_r:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_l2:
	RETROK_ESCAPE
[INFO] [Environ]: GET_VARIABLE vice_mapper_r2:
	RETROK_RETURN
[INFO] [Environ]: GET_VARIABLE vice_mapper_l3:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_r3:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_lr:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_ll:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_ld:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_lu:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_rr:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_rl:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_rd:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_ru:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_vkbd:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_statusbar:
	RETROK_F12
[INFO] [Environ]: GET_VARIABLE vice_mapper_joyport_switch:
	N/A
[INFO] [Environ]: GET_VARIABLE vice_mapper_reset:
	RETROK_END
[INFO] [Environ]: GET_VARIABLE vice_mapper_aspect_ratio_toggle:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_zoom_mode_toggle:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_warp_mode:
	---
[INFO] [Environ]: GET_VARIABLE vice_datasette_hotkeys:
	disabled
[INFO] [Environ]: GET_VARIABLE vice_mapper_datasette_toggle_hotkeys:
	---
[INFO] [Environ]: GET_VARIABLE vice_mapper_datasette_stop:
	RETROK_DOWN
[INFO] [Environ]: GET_VARIABLE vice_mapper_datasette_start:
	RETROK_UP
[INFO] [Environ]: GET_VARIABLE vice_mapper_datasette_forward:
	RETROK_RIGHT
[INFO] [Environ]: GET_VARIABLE vice_mapper_datasette_rewind:
	RETROK_LEFT
[INFO] [Environ]: GET_VARIABLE vice_mapper_datasette_reset:
	---
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [Environ]: RETRO_ENVIRONMENT_SET_CORE_OPTIONS_DISPLAY.
[INFO] [CONTENT LOAD]: Content loading skipped. Implementation will load it on its own.
[INFO] Version of libretro API: 1
[INFO] Compiled against API: 1
[INFO] [Audio]: Set audio input rate to: 48000.00 Hz.
[INFO] [Video]: Timings deviate too much. Will not adjust. (Display = 0.00 Hz, Game = 60.28 Hz)
[INFO] [Video]: Game FPS > Monitor FPS. Cannot rely on VSync.
[INFO] [Video]: Video @ 384x224
[INFO] [EGL] Falling back to eglGetDisplay
[INFO] [EGL]: EGL version: 1.4
[INFO] [GL]: Found GL context: fbdev_mali
[INFO] [GL]: Detecting screen resolution 0x0.
[INFO] [EGL]: Current context: 0x40000001.
[INFO] [GL]: Vendor: ARM, Renderer: Mali-400 MP.
[INFO] [GL]: Version: OpenGL ES 2.0.
[INFO] [GL]: Using resolution 384x224
[INFO] [GL]: Default shader backend found: glsl.
[INFO] [Shader driver]: Using GLSL shader backend.
[WARN] [GL]: Stock GLSL shaders will be used.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GLSL]: Found GLSL vertex shader.
[INFO] [GLSL]: Found GLSL fragment shader.
[INFO] [GLSL]: Linking GLSL program.
[INFO] [GL]: Using 4 textures.
[INFO] [GL]: Loaded 1 program(s).
[INFO] [Font]: Using font rendering backend: bitmap.
[INFO] [Video]: Graphics driver did not initialize an input driver. Attempting to pick a suitable driver.
[INFO] [Joypad]: Found joypad driver: "sdl".
[INFO] [Video]: Found display server: null
[INFO] ALSA: Using floating point format.
[INFO] ALSA: Period size: 512 frames
[INFO] ALSA: Buffer size: 1536 frames
[INFO] [Display]: Found display driver: "gl".
[ERROR] [Font]: Failed to create rendering backend: stb-unicode.
[INFO] [Font]: Using font rendering backend: bitmap.
[ERROR] [Font]: Failed to create rendering backend: stb-unicode.
[INFO] [Font]: Using font rendering backend: bitmap.
[ERROR] [Font]: Failed to create rendering backend: stb-unicode.
[INFO] [Font]: Using font rendering backend: bitmap.
[libretro DEBUG] retro_set_environment: GET_CORE_OPTIONS_VERSION failed, not setting CORE_OPTIONS now.
[libretro DEBUG] retro_set_environment: GET_CORE_OPTIONS_VERSION failed, not setting CORE_OPTIONS now.
[ERROR] [MIDI]: snd_seq_open failed with error -2.
[ERROR] [MIDI]: Initialization failed.
[INFO] [SRAM]: SRAM will not be saved.
[INFO] [Playlist]: Loading history file: [/.config/retroarch/content_history.lpl].
[INFO] [Playlist]: Loading history file: [/.config/retroarch/content_music_history.lpl].
[INFO] [Playlist]: Loading history file: [/.config/retroarch/content_image_history.lpl].
[INFO] [Playlist]: Loading favorites file: [/.config/retroarch/content_favorites.lpl].
[INFO] [GL]: VSync => OFF
[INFO] [EGL]: eglSwapInterval(0)
[INFO] [Playlist]: Written to playlist file: /.config/retroarch/content_history.lpl
[libretro INFO] Arg0: xvic
[libretro INFO] Arg1: Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].prg
[libretro INFO] No configuration file found at '/storage/savefiles/Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].vicerc'.
[libretro INFO] No configuration file found at '/storage/savefiles/vicerc'.
[libretro INFO] No configuration file found at '/storage/system/vice/vicerc'.
[libretro INFO] --------------------------------------------------------------------------------
[libretro INFO] VICE 3.5
[libretro INFO] Build date: May  7 2021 16:22:26
[libretro INFO] 
[libretro INFO] Welcome to xvic, the free portable VIC20 Emulator.
[libretro INFO] 
[libretro INFO] This is free software with ABSOLUTELY NO WARRANTY.
[libretro INFO] --------------------------------------------------------------------------------
[libretro INFO] VIC20 kernal patched to 901486-07.
[libretro INFO] DriveROM: 2000 ROM image not found. Hardware-level 2000 emulation is not available.
[libretro INFO] DriveROM: 4000 ROM image not found. Hardware-level 4000 emulation is not available.
[libretro INFO] DriveROM: CMDHD ROM image not found. Hardware-level CMDHD emulation is not available.
[libretro INFO] Drive: Finished loading ROM images.
[libretro INFO] Sound: Available sound devices: retro
[libretro INFO] Set resource: SoundFragmentSize => 0
[libretro INFO] Set resource: Mouse => 1
[libretro INFO] Set resource: Printer4 => 1
[libretro INFO] Set resource: AutostartPrgMode => 1
[libretro INFO] Set resource: AutostartDelayRandom => 0
[libretro INFO] Set resource: FileSystemDevice8 => 0
[libretro INFO] Set resource: FSDevice8ConvertP00 => 0
[libretro INFO] Set resource: VICPaletteFile => "colodore_vic"
[libretro INFO] Set resource: VICExternalPalette => 1
[libretro INFO] Set resource: VICFilter => 0
[libretro INFO] Set resource: VICPALBlur => 0
[libretro INFO] Set resource: VICColorGamma => 2800
[libretro INFO] Set resource: VICColorTint => 1000
[libretro INFO] Set resource: VICColorSaturation => 1000
[libretro INFO] Set resource: VICColorContrast => 1000
[libretro INFO] Set resource: VICColorBrightness => 1000
[libretro INFO] Set resource: UserportJoy => 0
[libretro INFO] Set resource: AutostartWarp => 0
[libretro INFO] Set resource: DriveTrueEmulation => 0
[libretro INFO] Set resource: VirtualDevices => 1
[libretro INFO] Set resource: AttachDevice8Readonly => 0
[libretro INFO] Set resource: DriveSoundEmulationVolume => 400
[libretro INFO] Set resource: DriveSoundEmulation => 1
[libretro INFO] Set resource: DatasetteSound => 0
[libretro INFO] Set resource: VICAudioLeak => 0
[libretro INFO] Set resource: SidEngine => 0
[libretro INFO] Set resource: MegaCartNvRAMWriteBack => 1
[libretro INFO] Set resource: RAMBlock0 => 0
[libretro INFO] Set resource: RAMBlock1 => 0
[libretro INFO] Set resource: RAMBlock2 => 0
[libretro INFO] Set resource: RAMBlock3 => 0
[libretro INFO] Set resource: RAMBlock5 => 0
[libretro INFO] AUTOSTART: Autodetecting image type of 'Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].prg'.
[libretro ERROR] Filesystem Image: Cannot open file 'Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].prg'.
[libretro ERROR] Tape: Cannot open file 'Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].prg'
[libretro ERROR] AUTOSTART: Cannot open 'Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].prg'.
[libretro ERROR] AUTOSTART: 'Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].prg' is not a valid file.
[libretro ERROR] Failed to autostart 'Aardvark Attack (1983-10)(Compute!)[Compute!'s Gazette][type-in].prg'
[libretro INFO] Main CPU: starting at ($FFFC).
[libretro INFO] Main CPU: RESET.

--- a/libretro/libretro-core.c
+++ b/libretro/libretro-core.c
@@ -774,12 +774,14 @@ static int process_cmdline(const char* argv)
              * addresses require special care for hassle-free usage. */
             if (path_is_valid(argv))
             {
+                char cart_base[RETRO_PATH_MAX] = {0};
                 char cart_2000[RETRO_PATH_MAX] = {0};
                 char cart_4000[RETRO_PATH_MAX] = {0};
                 char cart_6000[RETRO_PATH_MAX] = {0};
                 char cart_A000[RETRO_PATH_MAX] = {0};
+
+                char cartmega_base[RETRO_PATH_MAX] = {0};
                 char cartmega_nvram[RETRO_PATH_MAX] = {0};
-                char cartmega_temp[RETRO_PATH_MAX] = {0};
 
                 int type = vic20_autodetect_cartridge_type(argv);
 
@@ -789,6 +791,7 @@ static int process_cmdline(const char* argv)
                     FILE *fd;
                     char buf[RETRO_PATH_MAX] = {0};
                     char basepath[RETRO_PATH_MAX] = {0};
+                    char fullpath[RETRO_PATH_MAX] = {0};
                     snprintf(basepath, sizeof(basepath), "%s", argv);
                     path_basedir(basepath);
 
@@ -796,11 +799,12 @@ static int process_cmdline(const char* argv)
                     if (fgets(buf, sizeof(buf), fd) != NULL)
                     {
                         buf[strcspn(buf, "\r\n")] = 0;
-                        snprintf(basepath, sizeof(basepath), "%s%s", basepath, buf);
+                        snprintf(fullpath, sizeof(fullpath), "%s%s", basepath, buf);
                     }
                     fclose(fd);
 
-                    argv = basepath;
+                    argv = fullpath;
+                    type = vic20_autodetect_cartridge_type(argv);
                 }
 
                 switch (type)
@@ -835,11 +839,10 @@ static int process_cmdline(const char* argv)
                         Add_Option("-cartgeneric");
                         break;
                     case CARTRIDGE_VIC20_MEGACART:
-                        snprintf(cartmega_temp, sizeof(cartmega_temp), "%s", argv);
-                        snprintf(cartmega_temp, sizeof(cartmega_temp), "%s", path_basename(cartmega_temp));
-                        path_remove_extension(cartmega_temp);
+                        snprintf(cartmega_base, sizeof(cartmega_base), "%s", path_basename(argv));
+                        path_remove_extension(cartmega_base);
                         snprintf(cartmega_nvram, sizeof(cartmega_nvram), "%s%s%s%s",
-                                 retro_save_directory, FSDEV_DIR_SEP_STR, cartmega_temp, ".nvr");
+                                 retro_save_directory, FSDEV_DIR_SEP_STR, cartmega_base, ".nvr");
                         Add_Option("-mcnvramfile");
                         Add_Option(cartmega_nvram);
                         Add_Option("-cartmega");
@@ -851,97 +854,78 @@ static int process_cmdline(const char* argv)
                         switch (type)
                         {
                             case -1: /* Gamebase */
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", argv);
-                                path_remove_extension(cart_2000);
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "-2000", ""));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "-4000", ""));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "-6000", ""));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "-a000", ""));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s%s%s", cart_2000, "-2000", ".prg");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
 
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", argv);
-                                path_remove_extension(cart_4000);
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "-2000", ""));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "-4000", ""));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "-6000", ""));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "-a000", ""));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s%s%s", cart_4000, "-4000", ".prg");
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "-2000", ""));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "-4000", ""));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "-6000", ""));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "-a000", ""));
 
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", argv);
-                                path_remove_extension(cart_6000);
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "-2000", ""));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "-4000", ""));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "-6000", ""));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "-a000", ""));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s%s%s", cart_6000, "-6000", ".prg");
-
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", argv);
-                                path_remove_extension(cart_A000);
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "-2000", ""));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "-4000", ""));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "-6000", ""));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "-a000", ""));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s%s%s", cart_A000, "-a000", ".prg");
+                                snprintf(cart_2000, sizeof(cart_2000), "%s%s%s", cart_base, "-2000", ".prg");
+                                snprintf(cart_4000, sizeof(cart_4000), "%s%s%s", cart_base, "-4000", ".prg");
+                                snprintf(cart_6000, sizeof(cart_6000), "%s%s%s", cart_base, "-6000", ".prg");
+                                snprintf(cart_A000, sizeof(cart_A000), "%s%s%s", cart_base, "-a000", ".prg");
                                 break;
 
                             case -2: /* TOSEC */
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", argv);
-                                path_remove_extension(cart_2000);
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "[4000]", "[2000]"));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "[6000]", "[2000]"));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "[A000]", "[2000]"));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s%s", cart_2000, ".crt");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[4000]", "[2000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[6000]", "[2000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[A000]", "[2000]"));
+                                snprintf(cart_2000, sizeof(cart_2000), "%s%s", cart_base, ".crt");
 
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", argv);
-                                path_remove_extension(cart_4000);
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "[2000]", "[4000]"));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "[6000]", "[4000]"));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "[A000]", "[4000]"));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s%s", cart_4000, ".crt");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[2000]", "[4000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[6000]", "[4000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[A000]", "[4000]"));
+                                snprintf(cart_4000, sizeof(cart_4000), "%s%s", cart_base, ".crt");
 
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", argv);
-                                path_remove_extension(cart_6000);
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "[2000]", "[6000]"));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "[4000]", "[6000]"));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "[A000]", "[6000]"));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s%s", cart_6000, ".crt");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[2000]", "[6000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[4000]", "[6000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[A000]", "[6000]"));
+                                snprintf(cart_6000, sizeof(cart_6000), "%s%s", cart_base, ".crt");
 
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", argv);
-                                path_remove_extension(cart_A000);
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "[2000]", "[A000]"));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "[4000]", "[A000]"));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "[6000]", "[A000]"));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s%s", cart_A000, ".crt");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[2000]", "[A000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[4000]", "[A000]"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "[6000]", "[A000]"));
+                                snprintf(cart_A000, sizeof(cart_A000), "%s%s", cart_base, ".crt");
                                 break;
 
                             case -3: /* No-Intro */
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", argv);
-                                path_remove_extension(cart_2000);
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "$4000", "$2000"));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "$6000", "$2000"));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s", string_replace_substring(cart_2000, "$A000", "$2000"));
-                                snprintf(cart_2000, sizeof(cart_2000), "%s%s", cart_2000, ".20");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$4000", "$2000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$6000", "$2000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$A000", "$2000"));
+                                snprintf(cart_2000, sizeof(cart_2000), "%s%s", cart_base, ".20");
 
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", argv);
-                                path_remove_extension(cart_4000);
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "$2000", "$4000"));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "$6000", "$4000"));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s", string_replace_substring(cart_4000, "$A000", "$4000"));
-                                snprintf(cart_4000, sizeof(cart_4000), "%s%s", cart_4000, ".40");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$2000", "$4000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$6000", "$4000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$A000", "$4000"));
+                                snprintf(cart_4000, sizeof(cart_4000), "%s%s", cart_base, ".40");
 
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", argv);
-                                path_remove_extension(cart_6000);
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "$2000", "$6000"));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "$4000", "$6000"));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s", string_replace_substring(cart_6000, "$A000", "$6000"));
-                                snprintf(cart_6000, sizeof(cart_6000), "%s%s", cart_6000, ".60");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$2000", "$6000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$4000", "$6000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$A000", "$6000"));
+                                snprintf(cart_6000, sizeof(cart_6000), "%s%s", cart_base, ".60");
 
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", argv);
-                                path_remove_extension(cart_A000);
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "$2000", "$A000"));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "$4000", "$A000"));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s", string_replace_substring(cart_A000, "$6000", "$A000"));
-                                snprintf(cart_A000, sizeof(cart_A000), "%s%s", cart_A000, ".a0");
+                                snprintf(cart_base, sizeof(cart_base), "%s", argv);
+                                path_remove_extension(cart_base);
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$2000", "$A000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$4000", "$A000"));
+                                snprintf(cart_base, sizeof(cart_base), "%s", string_replace_substring(cart_base, "$6000", "$A000"));
+                                snprintf(cart_A000, sizeof(cart_A000), "%s%s", cart_base, ".a0");
                                 break;
                         }
 
@@ -1127,7 +1111,7 @@ static int process_cmdline(const char* argv)
      /* Disable floppy drive when using carts with cores that won't disable it via autostart */
      for (int i = 0; i < PARAMCOUNT; i++)
      {
-         if (strstr(XARGV[i], "-cart"))
+         if (strstr(XARGV[i], "-cart") && XARGV[i][0] == '-')
          {
              Add_Option("-drive8type");
              Add_Option("0");

